bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -141,60 +177,87 @@ final class BagsManager(executorService: ExecutorService) {
-                      LOG.warning(s"$bagsIdMsg failed: ${e.msg}")
-                      enqueue(obs, 5000L)
-
-                    // If we timed out, we don't want to delay.
-                    case Failure(ex: TimeoutException) =>
-                      LOG.warning(s"$bagsIdMsg failed: ${ex.getMessage}")
-                      enqueue(obs, 0L)
-
-                    // For all other exceptions, print the full stack trace.
-                    case Failure(ex) =>
-                      LOG.log(Level.WARNING, s"$bagsIdMsg} failed.", ex)
-                      enqueue(obs, 5000L)
+              if (isEligibleForBags(ctx)) {
+                // Note that we perform these checks AFTER eligibility since they are much more computationally
+                // intensive, and we need to store the hash value of hasBeenUpdated.
+                val (updated, hash) = hasBeenUpdated(obs, ctx)
+                if (updated && notObserved(obs)) {
+                  //   do the lookup
+                  //   on success {
+                  //      if we're in the queue again, it means something changed while this task was
+                  //      running, so discard this result and do nothing,
+                  //      otherwise update the model
+                  //   }
+                  //   on failure enqueue again, maybe with a delay depending on the failure
+                  val bagsIdMsg = s"BAGS lookup on thread=${Thread.currentThread.getId} for observation=${obs.getObservationID}"
+                  LOG.info(s"Performing $bagsIdMsg.")
+
+                  val newStatus = bagsStatus(key).fold(BagsStatus.NewRunning)(_.toRunning)
+                  setStatus(key, newStatus.some)
+                  Thread.sleep(1000)
+
+                  AgsRegistrar.currentStrategy(ctx).foreach { strategy =>
+                    val fut = strategy.select(ctx, OT.getMagnitudeTable)
+                    fut onComplete {
+                      case Success(opt) =>
+                        // If this observation is once again in the queue, then something changed while this task
+                        // was running, so discard the result.
+                        if (!state.keys(key)) {
+                          LOG.info(s"$bagsIdMsg successful. Results=${opt ? "Yes" | "No"}.")
+                          applySwingResults(opt)
+                          setStatus(key, None)
+                          updateHash(obs, hash)
+                        }
+
+                      // Note that we have to sleep after reporting a bags failure, as otherwise, there is no time for
+                      // the failure status message to be read before we change back to pending.
+
+                      // We don't want to print the stack trace if the host is simply unreachable.
+                      // This is reported only as a GenericError in a CatalogException, unfortunately.
+                      case Failure(CatalogException((e: GenericError) :: _)) =>
```

Comment:
I think it would be a good idea in the future (no pun intended) to enhance catalog exceptions, but I don't think it is something pressing that needs to be done sooner rather than later.